test/metabase/sample_data_test.clj

```diff
@@ -0,0 +1,88 @@
+
+(defn- sample-database-db []
+  {:details (#'sample-data/db-details)
+   :engine  :h2
+   :name    "Sample Database"})
+
+(defmacro ^:private with-temp-sample-database-db
+  "Execute `body` with a temporary Sample Database DB bound to `db-binding`."
+  {:style/indent 1}
+  [[db-binding] & body]
+  `(mt/with-temp Database [db# (sample-database-db)]
+     (#'sample-data/sync-sample-database! db#)
+     (let [~db-binding db#]
+       ~@body)))
+
+(defn- table
+  "Get the Table in a `db` with `table-name`."
+  [db table-name]
+  (db/select-one Table :name table-name, :db_id (u/the-id db)))
+
+(defn- field
+  "Get the Field in a `db` with `table-name` and `field-name.`"
+  [db table-name field-name]
+  (db/select-one Field :name field-name, :table_id (u/the-id (table db table-name))))
+
+
+;;; ----------------------------------------------------- Tests ------------------------------------------------------
+
+(deftest sync-sample-database-test
+  (testing (str "Make sure the Sample Database is getting synced correctly. In prod, only metadata sync is performed")
+    (with-redefs [config/is-prod? false
+                  config/is-test? true]
+      (with-temp-sample-database-db [db]
+        (is (= {:description      "The name of the user who owns an account"
+                :database_type    "VARCHAR"
+                :semantic_type    :type/Name
+                :name             "NAME"
+                :has_field_values :search
+                :active           true
+                :visibility_type  :normal
+                :preview_display  true
+                :display_name     "Name"
+                :fingerprint      {:global {:distinct-count 2499
+                                            :nil%           0.0}
+                                   :type   {:type/Text {:percent-json   0.0
+                                                        :percent-url    0.0
+                                                        :percent-email  0.0
+                                                        :percent-state  0.0
+                                                        :average-length 13.532}}}
+                :base_type        :type/Text}
+               (-> (field db "PEOPLE" "NAME")
+                   ;; it should be `nil` after sync but get set to `search` by the auto-inference. We only set `list` in
+                   ;; sync and setting anything else is reserved for admins, however we fill in what we think should be
+                   ;; the appropiate value with the hydration fn
+                   (hydrate :has_field_values)
+                   (select-keys [:name :description :database_type :semantic_type :has_field_values :active :visibility_type
+                                 :preview_display :display_name :fingerprint :base_type]))))))
+    ;; `fingerprint` and `semantic_type` are nil in prod since DB analysis is done async
+    (with-redefs [config/is-prod? true
+                  config/is-test? false]
```

Comment:
This block is the only real addition to this file. The rest is renaming the namespace to mirror the non-test one.